EXPLAINER.md

```diff
@@ -1,89 +1,134 @@
+# Interface
+The exact interfaces involved here are yet to be decided. But at a high level
+what we're providing is several bits:
+
+1. A modernized version of the existing (but not really standardized)
+   [`Entry`](https://www.w3.org/TR/2012/WD-file-system-api-20120417/#idl-def-Entry)
+   API in the form of new (names TBD) `FileSystemFileHandle` and
+   `FileSystemDirectoryHandle` interfaces.
+2. A modernized version of the existing (and also not really standardized)
+   [`FileWriter`](https://dev.w3.org/2009/dap/file-system/file-writer.html#the-filewriter-interface)
+   interface.
+3. Various entry points to get a handle representing a limited view of the
+   native file system. I.e. either via a file picker, or to get access to
+   certain well known directories, or even to get access to the whole native
+   file system. Mimicing things such as chrome's
+   [`chrome.fileSystem.chooseEntry`](https://developer.chrome.com/apps/fileSystem#method-chooseEntry) API.
+
+
 # Example code
 
+```javascript
+// Show a file picker to open a file.
+const file_ref = await FileSystemFileHandle.choose({
+    type: 'open',
+    accepts: [{description: 'Images', extensions: ['jpg', 'gif', 'png']}],
+    suggestedStartLocation: 'pictures-library'
+});
+if (!file_ref) {
+    // User cancelled, or otherwise failed to open a file.
+    return;
+}
+
+// Read the contents of the file.
+const file_reader = new FileReader();
+file_reader.onload = () => {
+  // ...
+};
+// file() method can reject if site (no longer) has access to the file.
+file_reader.readAsArrayBuffer(await file_ref.file());
+
+// ...
+
+// Write changed contents back to the file. Rejects if file reference is not
+// writable.
+const file_writer = await file_ref.createWriter();
+await file_writer.write(new Blob(['foobar']));
+file_writer.seek(1024);
```

Comment:
(The level of detail might be too low, in which case please disregard.) Not sure seek should be synchronous. It must be async if we want to give success / failure feedback, and seeking beyond end of file is a failure.